src/PerfView/PerfViewData.cs

```diff
@@ -3388,6 +3388,447 @@ private void CreateCSVFile(List<AspNetRequest> requests, string csvFileName)
+                                {
+                                    // activityId matches but PID is different, this shouldn't happen?
+                                    log.WriteLine("AspNetCoreStats encountered an identical ActivityId between processes on a request/stop. Ignoring this one.");
+                                }
+
+                            }
+                            else
+                            {
+                                // request does not exist yet, create and track it
+                                request = new ANCHostingRequest(traceEvent);
+                                request.EndTimeRelativeMSec = traceEvent.TimeStampRelativeMSec;
+                                requests.Add(request.ActivityId, request);
+                            }
+
+                            break;
+
+                        case _eventUnhandledException:
+                            if (traceEvent.ActivityID != null && requests.TryGetValue(traceEvent.ActivityID, out request))
+                            {
+                                // request exists, sanity PID check and create if necessary
+                                if (request.ProcessID == traceEvent.ProcessID)
+                                {
+                                    request.HasUnhandledException = true;
+                                }
+                                else
+                                {
+                                    // activityId matches but PID is different, this shouldn't happen?
+                                    log.WriteLine("AspNetCoreStats encountered an identical ActivityId between processes on an UnhandledException event. Ignoring this one.");
+                                }
+                            }
+                            else
+                            {
+                                otherEvents.Add(new OtherHostingEvent(traceEvent));
+                            }
+                            break;
+
+                        case _eventManifestData:
+                            // ignore
+                            break;
+
+                        case _eventHostStart:
+                        case _eventHostStop:
+                        case _eventServerReady:
+                        default:
+                            otherEvents.Add(new OtherHostingEvent(traceEvent));
+                            break;
+                    }
+                }
+                else continue;
+
+            }
+
+            // fixup requests without a start or a stop
+            if (requests.Count > 0)
+            {
+                foreach (ANCHostingRequest request in requests.Values)
+                {
+                    // DurationMsec only returns -1 if the request is not complete (i.e. does not have both a start and a stop event associated with it)
+                    if (request.DurationMsec == -1)
+                    {
+                        if (request.HasStart)
```

Comment:
Recommend flagging these incomplete requests in the UI somehow so that it's clear to users how the latency was calculated.  This is especially useful in the case of lost events where it might be confusing and cause folks to go after a long request that wasn't actually long.

Reply:
redid some of this output and also have some screenshots of the latest commit with some examples of this